modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/configuration.jsp

```diff
@@ -165,6 +165,7 @@ DLPortletInstanceSettingsHelper dlPortletInstanceSettingsHelper = new DLPortletI
 								<portlet:param name="mvcRenderCommandName" value="/document_library/select_folder" />
 								<portlet:param name="folderId" value="<%= dlAdminDisplayContext.isRootFolderInTrash() ? String.valueOf(DLFolderConstants.DEFAULT_PARENT_FOLDER_ID) : String.valueOf(dlAdminDisplayContext.getRootFolderId()) %>" />
 								<portlet:param name="ignoreRootFolder" value="<%= Boolean.TRUE.toString() %>" />
+								<portlet:param name="rootFolderInTrash" value="<%= String.valueOf(dlAdminDisplayContext.isRootFolderInTrash()) %>" />
```

Comment:
Instead of passing this as a parameter, why can't we use the same logic as in the other cases?
```
boolean rootFolderInTrash = false;

if (folderId != ...) {
    Folder folder = ...;

    if (folder.isRepositoryCapabilitySupported(...)) {
        TrashCapability trashCapability = folder.getRepositoryCapability(TrashCapability.class);
        rootFolderInTrash = trashCapability.isInTrash(folder);
    }
}
```
Not sure which of the two IDs is the correct one, though (folderId or selectedFolderId).

Also, make sure to check the TrashCapability is available in that repository with `folder.isCapabilitySupported`. Otherwise, this could throw an exception.

Reply:
I'll check that and resend. Thanks!

Reply:
In `select_folder.jsp` I have no information about the root folder that's in the trash: `folder` is `null` and `folderId` and `selectedFolderId` are `0` when I first try to select a new root folder. These values also happen when the root folder is actually deleted, not just in the trash, and also when the home folder is the root folder. 

To distinguish these three cases I need to pass the parameter `"rootFolderInTrash"` (and a new one `"rootFolderNotFound"` for deleted root folders, which I hadn't taken into account).

See new pr: #2348